modules/json/src/lib/parser/streaming-json-parser.js

```diff
@@ -35,12 +40,13 @@ export default class StreamingJSONParser extends JSONParser {
   }
 
   // PRIVATE METHODS
-
   _extendParser() {
     // Redefine onopenarray to locate top-level array
     this.parser.onopenarray = () => {
       if (!this.topLevelArray) {
         this.topLevelArray = [];
+        this.topLevelArrayPositions = [];
```

Comment:
This looks complicated. What is the big upside of this detailed cursor tracking?

Reply:
in csvLoading, `Papa.parse` has a `step` method that sends back `cursor` of current step, which is really nice for progress tracking. But `clarinet` doesn't. However, clarinet has an `position` property which is the position of the cursor. I added this so both the `parseJSONInBatches` and `parseCSVInBatches` can send a `cursor` when calling `tableBatchBuilder.addRow`